synfig-studio/src/gui/dialogs/dialog_pluginmanager.cpp

```diff
@@ -0,0 +1,252 @@
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+/* === H E A D E R S ======================================================= */
+
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include "dialog_pluginmanager.h"
+
+#include <fstream>
+
+#include <gtkmm/label.h>
+#include <gtkmm/notebook.h>
+#include <gtkmm/listbox.h>
+#include <gtkmm/listboxrow.h>
+
+#include <gui/localization.h>
+#include <gui/app.h>
+
+#include <synfig/general.h>
+#include <synfigapp/main.h>
+#include <synfig/os.h>
+
+#endif
+
+/* === U S I N G =========================================================== */
+
+using namespace synfig;
+using namespace studio;
+using namespace synfigapp;
+
+/* === M A C R O S ========================================================= */
+
+/* === G L O B A L S ======================================================= */
+
+/* === P R O C E D U R E S ================================================= */
+
+/* === M E T H O D S ======================================================= */
+
+Dialog_PluginManager::Dialog_PluginManager(Gtk::Window& parent):
+    Gtk::Dialog(_("Plugin Manager"), parent),
+    plugin_file_dialog("Select plugin zip file", Gtk::FILE_CHOOSER_ACTION_OPEN),
+    message_dialog(_("Are you sure you want to delete this plugin?"), false, Gtk::MESSAGE_QUESTION, Gtk::BUTTONS_OK_CANCEL, true),
+    plugin_list(App::plugin_manager.plugins())
+{
+    plugin_file_dialog.add_button("_Cancel", Gtk::RESPONSE_CANCEL);
+    plugin_file_dialog.add_button("_Open", Gtk::RESPONSE_OK);
+
+    // Add a filter to show only text files
+    auto filter_zip = Gtk::FileFilter::create();
+    filter_zip->set_name("Zip File (A synfig plugin zip file)");
+    filter_zip->add_mime_type("application/zip");
+    plugin_file_dialog.add_filter(filter_zip);
+
+    App::plugin_manager.signal_list_changed().connect(sigc::mem_fun(*this, &Dialog_PluginManager::refresh));
+    // this->set_resizable(false);
+    message_dialog.set_transient_for(*this);
+    Gtk::Button* install_plugin_button = manage(new Gtk::Button(_("Install Plugin")));
+    install_plugin_button->set_image_from_icon_name("document-open", Gtk::ICON_SIZE_BUTTON);
+    install_plugin_button->set_always_show_image(true);
+    install_plugin_button->signal_clicked().connect(sigc::mem_fun(*this, &Dialog_PluginManager::on_install_plugin_button_clicked));
+
+    Gtk::VBox* plugin_tab = manage(new Gtk::VBox());
+    plugin_tab->property_expand().set_value(false);
+    plugin_list_box.set_selection_mode(Gtk::SELECTION_NONE);
+
+    build_listbox();
+
+    plugin_tab->pack_start(plugin_list_box, Gtk::PACK_SHRINK);
+    plugin_tab->pack_end(*install_plugin_button, Gtk::PACK_SHRINK);
+
+    Gtk::Label* custom_label = manage(new Gtk::Label());
+    custom_label->set_text("Installed Plugins");
```

Comment:
Allow localization: surround translatable strings with `_()` (defined in gui/localization.h)